qt-bugs@ issue :  none
Trolltech task ID : none
bugs.kde.org number :
applied: no
author: Dirk Mueller

fix various obvious memory leaks etc

--- qt4-x11-4.3.2.orig/src/gui/painting/qimagescale.cpp
+++ qt4-x11-4.3.2/src/gui/painting/qimagescale.cpp
@@ -1020,6 +1020,7 @@
     buffer = QImage(dw, dh, src.format());
     if (buffer.isNull()) {
         qWarning("QImage: out of memory, returning null");
+        qimageFreeScaleInfo(scaleinfo);
         return QImage();
     }
 
--- qt4-x11-4.3.2.orig/src/svg/qsvghandler.cpp
+++ qt4-x11-4.3.2/src/svg/qsvghandler.cpp
@@ -1144,15 +1144,9 @@
         if (!size.isEmpty()) {
             QSvgHandler::LengthType type;
             qreal len = parseLength(size, type, handler);
-            //len = convertToPixels(len, false, type);
             // ### org_module.svg shows that font size
             // seems to be always in px...
-            type  = QSvgHandler::PX;
-            if (type == QSvgHandler::PX ||
-                type == QSvgHandler::OTHER)
-                font.setPixelSize(int(len));
-            else
-                font.setPointSizeF(len);
+            font.setPixelSize(int(len));
         }
         if (!style.isEmpty()) {
             if (style == QLatin1String("normal")) {
